File: 520. Detect Capital/520.py
class Solution(object):

    def detectCapitalUse(self, word):
        """
        :type word: str
        :rtype: bool
        """
        return not not list(filter(re.compile("([A-Z]*|[A-Z{1}][a-z]*|[a-z]*)$").match, [word]))

File: 0520. Detect Capital/520.cpp
class Solution {
public:
    bool detectCapitalUse(string word) {
        int ct = 0;
        for (int i=0; i<word.size(); ++i)
            ct += (word[i]<'a') ? 1 : 0;
        return (!ct || ct==word.size()) ? true : ((ct==1 && word[0]<'a') ? true : false);
    }
};